File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        minPrice = prices[0]
        maxProfit = 0

        for price in prices[1:]:
            profit = price - minPrice
            maxProfit = max(maxProfit, profit)
            minPrice = min(minPrice, price)  # update lowest price

        return maxProfit

File: Data Structures & Algorithms/best-time-to-buy-and-sell-stock-ii/submission-0.js
class Solution {
    /**
     * @param {number[]} prices
     * @return {number}
     */
    maxProfit(prices) {
        let profit = 0

        for (let i = 1; i < prices.length; i++) {
            if (prices[i] > prices[i - 1]) {
                profit += (prices[i] - prices[i - 1])
            }
        }
        return profit
    }
}
// If buy > sell advance buy 
// Sell is greater the buy we will buy the stock and sell it on that given day and we'll advance the buy pointer

